### DISCUSSION TITLE
Need isolate usage example with supabase

### DISCUSSION THREAD

**Discussion by richard457** [powersync-ja/powersync.dart#178] [Q&A]
Need isolate usage example with supabase

**Comment by rkistner** [comment]
From other conversations I gather:
1. You have some computation-heavy operations you want to do with the PowerSync database, so you want to run that in a background Isolate.
2. Supabase needs to run in the main Isolate (https://github.com/supabase/supabase-flutter/issues/1046).

So what I'd recommend:
1. Open and connect the `PowerSyncDatabase` in the main Isolate in the standard way, so all the Supabase-related logic will also be there.
2. Use `db.isolateConnectionFactory()` to create a connection in another Isolate.

Here's an example: https://gist.github.com/rkistner/e1067e51ad340f9447c4e55bc7bc96e1

The relevant bit for background isolate:

```dart
runInIsolate(IsolateConnectionFactory factory) async {
  return await Isolate.run(() async {
    var db = factory.open();
    var results = await db.getAll('select * from customers');
    return results;
  });
}

await runInIsolate(db.isolateConnectionFactory());
```

This example uses `Isolate.run` for a short-lived Isolate, but the same approach would work for long-lived Isolates using `Isolate.spawn`.

Note that inside the Isolate, the database is an instance of a sqlite_async `SqliteConnection`, rather than a full `PowerSyncDatabase`. All the query methods remain the same, and that should be all you need to work with the database.
